src/deps/uws.zig

```diff
@@ -784,95 +784,80 @@ extern fn us_socket_is_closed(ssl: i32, s: ?*Socket) i32;
     }
-    pub fn sendLastFragment(this: AnyWebSocket, message: []const u8, compress: bool) SendStatus {
-        switch (this) {
-            .tcp => return uws_ws_send_last_fragment(0, this.raw(), message.ptr, message.len, compress),
-            .ssl => return uws_ws_send_last_fragment(1, this.raw(), message.ptr, message.len, compress),
-        }
+
+    pub inline fn send(this: AnyWebSocket, message: []const u8, opcode: Opcode, compress: bool, fin: bool) SendStatus {
+        return uws_ws_send_with_options(this.is_ssl(), this.raw(), message.ptr, message.len, opcode, compress, fin);
     }
-    pub fn end(this: AnyWebSocket, code: i32, message: []const u8) void {
-        switch (this) {
-            .tcp => uws_ws_end(0, this.tcp.raw(), code, message.ptr, message.len),
-            .ssl => uws_ws_end(1, this.ssl.raw(), code, message.ptr, message.len),
-        }
+
+    pub inline fn sendLastFragment(this: AnyWebSocket, message: []const u8, compress: bool) SendStatus {
+        return uws_ws_send_last_fragment(this.is_ssl(), this.raw(), message.ptr, message.len, compress);
+    }
+
+    pub inline fn end(this: AnyWebSocket, code: i32, message: []const u8) void {
+        return uws_ws_end(this.is_ssl(), this.raw(), code, message.ptr, message.len);
     }
-    pub fn cork(this: AnyWebSocket, ctx: anytype, comptime callback: anytype) void {
+
+    pub inline fn cork(this: AnyWebSocket, ctx: anytype, comptime callback: anytype) void {
         const ContextType = @TypeOf(ctx);
         const Wrapper = struct {
             pub fn wrap(user_data: ?*anyopaque) callconv(.C) void {
                 @call(.always_inline, callback, .{bun.cast(ContextType, user_data.?)});
             }
         };
 
-        switch (this) {
-            .ssl => uws_ws_cork(1, this.raw(), Wrapper.wrap, ctx),
-            .tcp => uws_ws_cork(0, this.raw(), Wrapper.wrap, ctx),
-        }
+        return uws_ws_cork(this.is_ssl(), this.raw(), Wrapper.wrap, ctx);
     }
-    pub fn subscribe(this: AnyWebSocket, topic: []const u8) bool {
-        return switch (this) {
-            .ssl => uws_ws_subscribe(1, this.ssl.raw(), topic.ptr, topic.len),
-            .tcp => uws_ws_subscribe(0, this.tcp.raw(), topic.ptr, topic.len),
-        };
+
+    pub inline fn subscribe(this: AnyWebSocket, topic: []const u8) bool {
+        return uws_ws_subscribe(this.is_ssl(), this.raw(), topic.ptr, topic.len);
     }
-    pub fn unsubscribe(this: AnyWebSocket, topic: []const u8) bool {
-        return switch (this) {
-            .ssl => uws_ws_unsubscribe(1, this.raw(), topic.ptr, topic.len),
-            .tcp => uws_ws_unsubscribe(0, this.raw(), topic.ptr, topic.len),
-        };
+
+    pub inline fn unsubscribe(this: AnyWebSocket, topic: []const u8) bool {
+        return uws_ws_unsubscribe(this.is_ssl(), this.raw(), topic.ptr, topic.len);
     }
-    pub fn isSubscribed(this: AnyWebSocket, topic: []const u8) bool {
-        return switch (this) {
```

Comment:
these patterns were actually deliberate

wanted to tell LLVM that these are separate functions

Reply:
but i didn't inspect the LLVM IR to see if it actually did the monomorphization